### PR TITLE
Social Previews: Update package to support titleless posts

### PR DESCRIPTION
**Pull request by gmjuhasz**
Related to https://github.com/Automattic/jetpack/pull/35728

## Proposed Changes

* Updated each preview to support our new titleless Social Notes format.

## Testing Instructions

This cannot be tested in Calypso natively as the previews for social posts live in jetpack under wp-admin.

Test together with this change: https://github.com/Automattic/jetpack/pull/35728

For sanity testing, you can check that in Calypso under the Posts menu if you check the previews, it works as intended.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/social-previews/src/nextdoor-preview/link-preview.tsx [hunk 1]
```diff
@@ -15,6 +16,7 @@ export function NextdoorLinkPreview( props: NextdoorLinkPreviewProps ) {
 			// Override the props that are irrelevant to link preview
 			description=""
 			media={ undefined }
+			title={ props.title || stripHtmlTags( props.description ).substring( 0, 50 ) }
```
Comment (pablinos) [inline packages/social-previews/src/nextdoor-preview/link-preview.tsx]: I know it's not much, but do you think we should make this a helper function? If we want to change the length or something, we'll only have to change it in one place then.

**Review by pablinos — COMMENTED** [review]
This is working ok. I notice we're not passing the description on as the title for twitter though, which might be the problem I'm seeing.

File: packages/social-previews/src/helpers.tsx [hunk 2]
```diff
@@ -40,6 +40,15 @@ export const stripHtmlTags: Formatter< Array< string > > = ( description, allowe
 	return description ? description.replace( pattern, '' ) : '';
 };
 
+/**
+ * For social note posts we use the first 50 characters of the description.
+ * @param description The post description.
+ * @returns The first 50 characters of the description.
+ */
+export const getNoteTitle = ( description: string ): string => {
```
Comment (pablinos) [inline packages/social-previews/src/helpers.tsx]: This is pretty pedantic, but I wonder if this should be `getPreviewTitle` or `getTitleFromDescription` or something as there might be other title-less posts like the Aside post format.

**Review by pablinos — COMMENTED** [review]
I think we might need to check what we do if someone hasn't provided a title for a normal post, but clicks on the preview. Currently the title is falsey, so this new behaviour kicks in.

CleanShot 2024-02-19 at 20 46 43@2x

CleanShot 2024-02-19 at 20 47 14@2x

Is that accurate?

**Comment by gmjuhasz** [comment]
> I think we might need to check what we do if someone hasn't provided a title for a normal post, but clicks on the preview. Currently the title is falsey, so this new behaviour kicks in.
> 
> Is that accurate?

@pablinos well, it's hard to say what's accurate there, as they cannot share that normal post before giving a title. On the other hand, I think the preview without the title looks reasonable, but it does not really matter, as they have to set a title to share the post. Maybe we could even promote Social Notes there if they try to check the preview without a title!

**Review by pablinos — APPROVED** [review]
I don't like the nested ifs in the Mastodon component but we can refactor that when we update the app previews

**Pull request merged**